packages/hint-sri/src/hint.ts

```diff
@@ -340,139 +340,158 @@ Actual:   ${integrities.join(', ')}`;
+        if (this.scanEnded && isInCache) {
             /*
              * The item is cached. For the VSCode extension and the
              * local connector with option 'watch' activated we
              * should report what we have in the cache after the
              * first 'scan::end'.
              */
-            if (this.scanEnded) {
-                const promises = this.getCache(evt).map((error) => {
-                    return this.context.report(error.resource, error.message, error.options);
-                });
+            const promises = this.getCache(evt).map((error) => {
+                return this.context.report(error.resource, error.message, error.options);
+            });
 
-                await Promise.all(promises);
-            }
+            await Promise.all(promises);
 
-            return;
+            // Return false to stop the validations.
+            return false;
         }
 
-        const { resource, element } = evt;
-        let content: NetworkData;
-
-        /*
-         * If the resource is a local file, ignore the analysis.
-         * The sri usually is added on the building process before publish,
-         * so is going to be very common that the sri doesn't exists
-         * for local files.
-         */
-        if (resource.startsWith('file://')) {
-            debug(`Ignoring local resource: ${resource}`);
+        return !isInCache;
+    }
 
-            return;
-        }
+    private async downloadContent(evt: FetchEnd) {
+        const { resource, response, element } = evt;
 
         /*
          * `requestAsync` is not included in webpack bundle for `extension-browser`.
          * This is ok because the browser will have already requested this via `fetch::end`
          * events.
          *
-         * Note: We are not using `Requester` becuase it depends on `iltorb` and it can
+         * Note: We are not using `Requester` because it depends on `iltorb` and it can
          * cause problems with the vscode-extension because `iltorb` dependens on the
          * node version for which it was compiled.
+         *
+         * We can probably use Requester once https://github.com/webhintio/hint/issues/1604 is done,
+         * and vscode use the node version that support it.
+         *
+         * When the crossorigin="use-credentials" but the response contains
+         * the header `Access-Control-Allow-Origin` with value `*` Chrome block the access
+         * to the resource by CORS policy, so we will reach this point
+         * through the traverse of the dom and response.body.content will be ''. In this case,
+         * we have to prevent download any resource.
          */
+        if (!requestAsync && !response.body.content) {
+            // Stop the validations.
+            return false;
+        }
+
         if (!requestAsync) {
-            return;
+            return true;
+        }
+
+        /*
+         * If the content already exists, we don't need to download it.
```

Comment:
Nit: You can collapse and use a single line comment `//` or `/* */` in the same line.